base.lua

```diff
@@ -1870,17 +1870,35 @@ LoadTrigger.getIsActivatable = Utils.overwrittenFunction(LoadTrigger.getIsActiva
 
 -- LoadTrigger doesn't allow filling non controlled tools
 function courseplay:onActivateObject(superFunc,vehicle)
-	if vehicle~= nil then
-		--if i'm in the vehicle, all is good and I can use the normal function, if not, i have to cheat:
-		if g_currentMission.controlledVehicle ~= vehicle then
-			local oldControlledVehicle = g_currentMission.controlledVehicle;
-			g_currentMission.controlledVehicle = vehicle;
-			superFunc(self);
-			g_currentMission.controlledVehicle = oldControlledVehicle;
-			return;
-		end
+	--No vehicle, non-courseplay call so just call default version
+	if vehicle == nil then
+		superFunc(self);
+		return;
+	end
+	--Vehicle present, called from courseplay
+	--Cache giant values to restore later
+	local defaultGetFarmIdFunction = g_currentMission.getFarmId;
+	local oldControlledVehicle = g_currentMission.controlledVehicle;
+
+	--Override farm id to match the calling vehicle (fixes issue when obtaining fill levels)
+	local overriddenFarmIdFunc = function()
+		local ownerFarmId = vehicle:getOwnerFarmId()
+		courseplay:debug(string.format('Overriding farm id during loading to %d', ownerFarmId), 19);
```

Comment:
Please use courseplay.debugVehicle() instead.

Reply:
Sorry, are you suggesting replacing the existing courseplay calls to LoadTrigger:onActivateObject with a different function which performs the necessary changes either side of the call instead?

Reply:
No, I'm suggesting not overriding LoadTrigger.onActivateObject instead create a new function, for instance 
courseplay.onActivateTrigger(trigger, vehicle)
which would just do the replacement of global variables and then call trigger:onActivateObject() then restore the globals. 

This way there's no need for
https://github.com/Courseplay/courseplay/blob/6ec33346b2e6ca87952c7b3c8076dbe35b94428d/base.lua#L1885

But again, this is optional.

Reply:
Great, we are in agreement then (it may not have been clear but thats what i was trying to describe). I'll look at making that change (reduces technical debt in the long run) and also the debug change.

Reply:
Now that I read it the second time I see we were talking about the same thing :)

Special thanks for the 'reducing technical debt', we have a huge amount of that in CP...

Reply:
Haha yes it is inevitable I feel. I've pushed the refactor now.